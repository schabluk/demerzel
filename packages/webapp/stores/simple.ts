import { types } from 'mobx-state-tree'

// Model definition.
export const ListItem = types.model('ListItem', {
  id: types.number,
  name: types.string,
})

// Store definition.
export const ListStore = types
  .model('SimpleStore', {
    items: types.optional(types.array(ListItem), []),
  })
  .views(self => ({
    get size() {
      return self.items.length
    },
  }))
// .actions(self => {
//   return {
//     afterCreate() {
//       console.log('SimpleStore afterCreate', getSnapshot(self))
//     },
//   }
// })
