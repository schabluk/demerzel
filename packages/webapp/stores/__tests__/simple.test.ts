import { getSnapshot } from 'mobx-state-tree'
import { ListStore } from '../simple'

const snapshot = {
  items: [{ id: 0, name: 'Foo' }, { id: 1, name: 'Bar' }],
}

describe('Store initialization', () => {
  it('Initialize with empty list', () => {
    const list = ListStore.create({})

    expect(list.size).toBe(0)
  })

  it('Initialize with two items', () => {
    const list = ListStore.create(snapshot)

    expect(list.size).toBe(2)
  })
})

it('Matches the file snapshot', () => {
  const list = ListStore.create(snapshot)

  expect(getSnapshot(list)).toMatchSnapshot()
})

it('Matches the inline snapshot', () => {
  const list = ListStore.create(snapshot)

  expect(getSnapshot(list)).toMatchInlineSnapshot(`
Object {
  "items": Array [
    Object {
      "id": 0,
      "name": "Foo",
    },
    Object {
      "id": 1,
      "name": "Bar",
    },
  ],
}
`)
})

test.todo('write more test cases')
