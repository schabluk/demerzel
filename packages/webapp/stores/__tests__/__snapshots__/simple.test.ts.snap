// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the file snapshot 1`] = `
Object {
  "items": Array [
    Object {
      "id": 0,
      "name": "Foo",
    },
    Object {
      "id": 1,
      "name": "Bar",
    },
  ],
}
`;
